mlflow/exceptions.py

```diff
@@ -143,4 +143,7 @@ class InvalidUrlException(MlflowException):
 class UnsupportedMultipartUploadException(MlflowException):
     """Exception thrown when multipart upload is unsupported by an artifact repository"""
 
-    pass
+    message = "Multipart upload is not supported for the current artifact repository"
+
+    def __init__(self):
+        super().__init__(self.message, error_code=NOT_IMPLEMENTED)
```

Comment:
changed this to 501 response which should make more sense. Also, removing the artifact repo class name from the message.

Reply:
sg!